### ISSUE TITLE
Tuple entity is taken by the type of the repository, and not by the type of return of the function

### ISSUE THREAD

**Issue by ArtDu** [tarantool/cartridge-springdata#32]
```java
public interface PetRepository extends TarantoolRepository<Pet, Integer> {

	/**
	 * Retrieve all {@link PetType}s from the data store.
	 * @return a Collection of {@link PetType}s.
	 */
//	@Query("SELECT ptype FROM PetType ptype ORDER BY ptype.name")
//	@Transactional(readOnly = true)
	@Query(function = "find_pet_types")
	List<PetType> findPetTypes();
```

```lua
local function find_pet_types()
    return crud.select("types")
end
```

![image](https://user-images.githubusercontent.com/22258443/110444327-ef842d00-80cd-11eb-8d7b-63c1e4e7ea87.png)

```
2021-03-09 11:38:19.759 ERROR 46506 --- [nio-8080-exec-8] o.a.c.c.C.[.[.[/].[dispatcherServlet]: Servlet.service() for
 servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is
org.springframework.core.convert.ConversionFailedException: Failed to convert from type [java.util.ArrayList<?>] to type
[@org.springframework.data.tarantool.repository.Query java.util.List<org.springframework.samples.petclinic.owner.PetType>] for value '[cat, dog, lizard, snake, bird, hamster]'; 
nested exception is org.springframework.core.convert.ConverterNotFoundException: No converter found capable of 
converting from type [org.springframework.samples.petclinic.owner.Pet] to type 
[@org.springframework.data.tarantool.repository.Query org.springframework.samples.petclinic.owner.PetType]] with root 
cause

org.springframework.core.convert.ConverterNotFoundException: No converter found capable of converting from type
 [org.springframework.samples.petclinic.owner.Pet] to type [@org.springframework.data.tarantool.repository.Query org.springframework.samples.petclinic.owner.PetType]
```

**Comment by vrogach2020** [comment]
Also the following scenario must be fixed.
Tarantool side function:
```lua
function test_custom(a, b)
  return tostring(a + b)
end
```

Repository method definition
```java
    @Query(function = "test_custom")
    String testCustom(Integer a, Integer b);
```

Repository usage
```java
    @Test
    public void testCustomMethod() {
        String r = bookRepository.testCustom(1,2);
        assertEquals("3", r);
    }
```

Now it throws the following error:
```
org.springframework.dao.InvalidDataAccessResourceUsageException: The received result cannot be converted to an array of tuples: 3; nested exception is io.tarantool.driver.exceptions.TarantoolClientException: The received result cannot be converted to an array of tuples: 3
```

**Comment by ArtDu** [comment]
The workaround, for now, is to use the wrapper

```java
public class StringWrapper {
    private String string;
}
```

```java
public interface StringRepository extends TarantoolRepository<StringWrapper, Long> {
    @Query(function = "custom_function")
    StringWrapper customFunction();
}
```

```java
@Test
public void test() {
    String s = stringRepository.customFunction();
    assertEquals("hello", s);
}
```

```lua
function custom_function()
    return {string = "hello"}
end
```

**Comment by ArtDu** [comment]
Closes by #48
